src/expr/datatype.cpp

```diff
@@ -1350,7 +723,26 @@ Type DatatypeConstructorArg::getRangeType() const {
 
 bool DatatypeConstructorArg::isUnresolvedSelf() const
 {
-  return d_selector.isNull() && d_name.size() == d_name.find('\0') + 1;
+  std::string name = getName();
+  return d_selector.isNull() && name.size() == name.find('\0') + 1;
+}
+
+bool DatatypeConstructorArg::isResolved() const
+{
+  // We could just write:
+  //
+  //   return !d_selector.isNull() && d_selector.getType().isSelector();
+  //
+  // HOWEVER, this causes problems in ExprManager tear-down, because
+  // the attributes are removed before the pool is purged.  When the
+  // pool is purged, this triggers an equality test between Datatypes,
+  // and this triggers a call to isResolved(), which breaks because
+  // d_selector has no type after attributes are stripped.
+  //
+  // This problem, coupled with the fact that this function is called
+  // _often_, means we should just use a boolean flag.
+  //
+  return d_internal->isResolved();
```

Comment:
Add `ExprManagerScope ems(d_constructor);` to be on the safe side.

Reply:
This would cause segfaults for reasons mentioned above.